runtime/metricsview/executor_rewrite_clickhouse_dicts.go

```diff
@@ -0,0 +1,91 @@
+package metricsview
+
+import (
+	"fmt"
+
+	"github.com/rilldata/rill/runtime/drivers"
+)
+
+// Rewrites filter on dictionary columns to use the key column instead of the value column. For example, if the dictionary value column is "country" and key column is "country_key",
+// then the filter will be rewritten to country_key = 'US' instead of country = 'USA'.
+func (e *Executor) rewriteClickhouseDictFilters(qry *Query) {
```

Comment:
Since lookups are introduced as a general concept in the metrics view spec, I wonder if it should be handled directly inside `astexpr.go` when converting an expression to SQL? 

I think that would also avoid the need for exposing internal fields in the `Query` type (which is supposed to be user-facing, as opposed to `AST` which is intended to be internal).

Reply:
I did it earlier as I was doing the group by optimization as well so thought it would be good to define both as standalone rewrite that be configured on/off.